stdlib/Random/src/normal.jl

```diff
@@ -137,6 +146,9 @@ end
     end
 end
 
+### fallback randexp for float types defining rand:
+randexp(rng::AbstractRNG, ::Type{T}) where {T<:AbstractFloat} =
+    -log(1-rand(rng, T))
```

Comment:
Should this use `log1p`?

Reply:
I guess it couldn't hurt.

Reply:
The downside of this is that new types are probably more likely to define `log` than `log1p`, but that is arguably just them being bad (since `log1p` is arguably the better function)

Reply:
Even if they define `log1p(x) = log(1+x)` as a stopgap, it's not so bad for `randexp`.